code/datums/components/atom_linker.dm

```diff
@@ -0,0 +1,82 @@
+/datum/component/atom_linker
+	dupe_mode = COMPONENT_DUPE_ALLOWED
+	var/id
+	var/list/linked_components
+	var/static/list/linker_lists = list()
+
+/datum/component/atom_linker/Initialize(id)
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE
+	
+	if(!id || (!istext(id) && !ispath(id)))
+		. = COMPONENT_INCOMPATIBLE
+		CRASH("Invalid link id: [id]")
+
+	src.id = id
+	linked_components = linker_lists[id]
+	if(!linked_components)
+		linker_lists[id] = linked_components = list()
+	linked_components += src
+
+/datum/component/atom_linker/Destroy()
+	linked_components -= src
+	linked_components = null
+	if(!linked_components.len)
+		linker_lists[id] = null
+	return ..()
+
+/datum/component/atom_linker/proc/GetLinks()
+	return linked_components.Copy()
+
+/atom/proc/GetLinkedAtoms(id)

```

Comment:
When would you call this already knowing what id to pass in, if the id is intended to be defined entirely by the component?